sources/ghostdog/src/main.rs

```
@@ -319,66 +368,29 @@ lazy_static! {
     ].iter().copied().collect();
 }
 
-// Returning a Result from main makes it print a Debug representation of the error, but with Snafu
-// we have nice Display representations of the error, so we wrap "main" (run) and print any error.
-// https://github.com/shepmaster/snafu/issues/110
-fn main() {
-    if let Err(e) = run() {
-        eprintln!("{}", e);
-        std::process::exit(1);
-    }
-}
+/// write_guid will write the provided guid in the environment variable format needed for
+/// starting services such as NVIDIA Fabric Manager or NVLSM to the requested path
+fn write_guid(config_path: PathBuf, guid: &str) -> Result<()> {
+    // Create a temporary file in the desired config directory
+    let env_directory = config_path.parent().unwrap_or(Path::new("/"));
```

This should return an error instead of defaulting to `/` as the parent directory.

---

Good catch. I've made it error when a suitable path isn't provided.